src/app/projectView.hook.js

```diff
@@ -53,12 +52,26 @@ export default (
         .then(() => {
           ProjectLoadingEmitter.emit(FINISHED_LOADING_PROJECT_EVENT);
         })
-        .catch(() => {
+        .catch(targetState => {
+          let stateService = trans.router.stateService;
+
+          if (targetState) {
+            let state = targetState._identifier;
+            // nextState and nextStateParams are executed from
+            // login.component.js after user logs in
+            return stateService.target(state, {
+              nextState: trans.to(checkProjectViewPresent),
+              nextStateParams: trans.to(checkProjectViewPresent).params,
```

Comment:
what is the reason for passing `checkProjectViewPresent` here?

Reply:
also, a cleaner way to do this where you aren't accessing "private" variable (denoted by the `_`) is to use the `withParams` method on the targetState.

```
return targetState.withParams({
    nextState: ...,
    ...
});
```

Reply:
I don't know how to access the state name without using _identifier. Please take a look and let me know.  I couldn't make targetState.withParams return the state name either.

Reply:
you don't need the state name.

Returning:
```
return targetState.withParams({
              nextState: trans.to(),
              nextStateParams: trans.to().params,
            });
```

should work just fine. It's essentially doing the same thing as 

```
return stateService.target(targetState._identifier, {
              nextState: trans.to(),
              nextStateParams: trans.to().params,
            });
```

except it is cleaner b/c you use the public api and don't have to access the sudo "private" `_identifier` property.

lmk if that doesn't work for some reason